src/main/kotlin/com/replaymod/gradle/preprocess/PreprocessPlugin.kt

```
@@ -346,4 +347,17 @@ private val Project.tinyMappings: File?
         throw GradleException("loom version not supported by preprocess plugin")
     }
 
+private val Task.classpath: FileCollection?
+    get() = if (this is AbstractCompile) {
+        this.classpath
+    } else {
+        // assume kotlin 1.7+
+        try {
+            val classpathMethod = this.javaClass.getMethod("getLibraries")
+            classpathMethod.invoke(this) as? FileCollection
```

Should this not be `as FileCollection?` or can that method return other types which we're supposed to silently ignore?

---

i always just wrote it as that instead of `as Class?` lol i didn't know those two were different things